tools/KoreBuild.Console/Commands/SubCommandBase.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            _koreBuildDir = FindKoreBuildDirectory();
+
+            base.Configure(application);
+
+            _toolsSourceOption = application.Option("--toolsSource", "The source to draw tools from.", CommandOptionType.SingleValue);
+            _repoPathOption = application.Option("--repoPath", "The path to the repo to work on.", CommandOptionType.SingleValue);
+            _dotNetHomeOption = application.Option("--dotNetHome", "The place where dotnet lives", CommandOptionType.SingleValue);
+        }
+
+        protected override bool IsValid()
+        {
+            if(!Directory.Exists(RepoPath))
+            {
+                System.Console.WriteLine($"The RepoPath '{RepoPath}' doesn't exist.");
+                return false;
+            }
+
+            return base.IsValid();
+        }
+
+        protected int RunDotnet(string[] arugments)
+        {
+            var args = ArgumentEscaper.EscapeAndConcatenate(arugments);
+
+            var psi = new ProcessStartInfo
+            {
+                FileName = GetDotNetExecutable(),
+                Arguments = args
+            };
+
+            var process = Process.Start(psi);
+            process.WaitForExit();
+
+            return process.ExitCode;
+        }
+
+        private string GetDotnetSDKVersion()
+        {
+            var sdkVersionEnv = Environment.GetEnvironmentVariable("KOREBUILD_DOTNET_VERSION");
+            if (sdkVersionEnv != null)
+            {
+                return sdkVersionEnv;
+            }
+            else
+            {
+                var sdkVersionPath = Path.Combine(ConfigDirectory, "sdk.version");
+                return File.ReadAllText(sdkVersionPath).Trim();
+            }
+        }
+
+        private string GetDotNetHome()
+        {
+            var dotnetHome = Environment.GetEnvironmentVariable("DOTNET_HOME");
+            var userProfile = Environment.GetEnvironmentVariable("USERPROFILE");
+            var home = Environment.GetEnvironmentVariable("HOME");
+
+            var dotnetFolderName = ".dotnet"; 
+
+            var result = Path.Combine(Directory.GetCurrentDirectory(), dotnetFolderName);
+            if (_dotNetHomeOption.HasValue())
+            {
+                result = _dotNetHomeOption.Value();
+            }
+            else if (!string.IsNullOrEmpty(dotnetHome))
+            {
+                result = dotnetHome;
+            }
+            else if (!string.IsNullOrEmpty(userProfile))
+            {
+                result = Path.Combine(userProfile, dotnetFolderName);
+            }
+            else if (!string.IsNullOrEmpty(home))
+            {
+                result = home;
+            }
+
+            return result;
+        }
+
+        private string FindKoreBuildDirectory()
+        {
+            var executingDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            var root = Directory.GetDirectoryRoot(executingDir);
+            while (true)
```

Comment:
It's usually better to put some kind of condition here. `while(true)` is a dangerous pattern.